### ISSUE TITLE
Consider not disabling Key Metrics selection accordion

### ISSUE THREAD

**Issue by nfmohit** [google/site-kit-wp#7383]
## Feature Description

For manual metric selection in Key Metrics, we allow selecting up to 4 metrics. We disable all remaining unchecked metrics when the user has selected 4 metrics. According to the [Figma designs](https://www.figma.com/file/CdtrHB0HAjfGRWTnTMMQI0/Key-Metrics?node-id=1974%3A61989&mode=dev), besides disabling the Checkbox, we also disable the Accordion that displays further information regarding the respective metric.

![image](https://github.com/google/site-kit-wp/assets/20284937/e552245c-0e52-43a3-aa0a-b2ee1544934c)

While working on #6258, I and @aaemnnosttv had [a discussion](https://github.com/google/site-kit-wp/pull/7326#discussion_r1275383591) where we felt it wasn't necessary to disable the accordion.

> The behavior of the accordion only toggles the visibility of the description which doesn't necessarily need to be disabled by the selection limits (only the checkboxes really need to be disabled). It could be useful to be able to see what the descriptions are without deselecting one to be able to read it.

Besides, it also creates a layout shift when an unchecked but accordion-opened gets closed when the user has checked 4 other metrics:

https://github.com/google/site-kit-wp/assets/20284937/de3fe83a-1782-4ab9-bd7a-3f6f1cf76e70

This issue is opened as a measure to reconsider if we really want to keep the accordion disabled in this scenario.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by sigal-teller** [comment]
@jimmymadon Yes well if the collapse will work it shouldn't be grayed out, and clicking anywhere (besides the checkbox) should open the accordion. It's not a great visibility just to have the checkbox grayed out, I'm not sure users will understand that it's disabled if the entire listing is not grayed out, but we don't want to get in redesigning things here, because in the next version that will include events and conversions we'll use a slightly different UX pattern so no metric will be grayed out at all. So don't put to mush effort in this now, just do what you suggested.

**Comment by techanvil** [comment]
AC :white_check_mark:

**Comment by techanvil** [comment]
Hey @nfmohit, this IB is looking good. There's one more aspect that needs to be considered though.

Currently, `children` is a required prop for the `Checkbox`, so as we're going to remove `children` from the rendered `Checkbox` instances, we should also make corresponding changes to the `Checkbox` components (both the GM2 and GM3 versions to keep them in sync).

So, `children` should no longer be a required prop in these components, and anything within them that relies on `children` (i.e. the label-related aspects) should also be updated to accommodate the case where the prop is not supplied.

**Comment by nfmohit** [comment]
That is a very important concern, thank you for raising, @techanvil! I have updated the IB to address this, and also added some considerations regarding accessibility. Please let me know how it looks now, thanks!

**Comment by techanvil** [comment]
Thanks @nfmohit, that looks great, nice work.

IB :white_check_mark:

**Comment by bethanylang** [comment]
@aaemnnosttv @jimmymadon Given that we're going to disable the accordion behavior prior to launch, is it okay to close this one out as `wontfix`?
